ncwms_configurator/ncwms_configurator.py

```diff
@@ -0,0 +1,219 @@
+import sys
+import os
+import logging
+from argparse import ArgumentParser
+
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from modelmeta import DataFile, DataFileVariable, Ensemble, EnsembleDataFileVariables
+
+from lxml import etree
+
+log = logging.getLogger(__name__)
+
+def get_variable(id, title, colorScaleRange, palette="rainbow", scaling="linear", numColorBands="250", disabled="false"):
+    return etree.Element(
+        "variable",
+        id = id,
+        title = title,
+        colorScaleRange = colorScaleRange,
+        palette = palette,
+        scaling = scaling,
+        numColorBands = numColorBands,
+        disabled = disabled
+    )
+
+
+def get_dataset(id, location, title, queryable="true", dataReaderClass="", copyrightStatement="", moreInfo="", disabled="false", updateInterval="-1"):
+    return etree.Element(
+        "dataset",
+        id = id,
+        location = location,
+        title = title,
+        queryable = queryable,
+        dataReaderClass = dataReaderClass,
+        copyrightStatement = copyrightStatement,
+        moreInfo = moreInfo,
+        disabled = disabled,
+        updateInterval = updateInterval
+    )
+
+
+
+
+def get_contact(name = "", organization = "", telephone = "", email = ""):
+    root = etree.Element("contact")
+    etree.SubElement(root, "name").text = name
+    etree.SubElement(root, "organization").text = organization
+    etree.SubElement(root, "telephone").text = telephone
+    etree.SubElement(root, "email").text = email
+    return root
+
```

Comment:
All of these `get_*()` functions are very clear in what they do, but they're a bit verbose and repetitive. For example, `get_contact()` could as easily be:

```python
def get_contact(**kwargs):
    root = etree.Element("contact")
    for k, v in kwargs:
        etree.SubElement(root, k).text = v
    return root
```

And you can further generalize that by adding the element type as a parameter and then it covers `get_server()` and `get_cache()` as well. E.g.

```python
def get_element(element_name, **kwargs):
    root = etree.Element(element_name)
    ...
```

Downside is that the required keyword args aren't explicit, but that could easily be included.

It might be worth the time to refactor these functions in that way. What do you think?